### PR TITLE
Add customer route

### PR DESCRIPTION
**Pull request by hibes**

### CODE REVIEW

File: src/lib/send-resolution.js [hunk 1]
```diff
@@ -4,8 +4,9 @@ const parseResponse = require('./parse-response');
 
 // The function should promise an object with a response (response for text or jsonRespone for json)
 // and an optional status code (200 OK is assumed if absent)
-module.exports = (context, fnWillPromise) => (req, res, next) =>
-  fnWillPromise(req, res)
+module.exports = (context, fnMayPromise) => (req, res, next) =>
```
Comment (geoff-sheldon) [inline src/lib/send-resolution.js]: Currying!  Had to do some reading on this, makes sense now :)

File: src/routes.js [hunk 2]
```diff
@@ -10,4 +10,16 @@ module.exports = context => [
       ),
     ),
   },
+  {
+    route: '/customer',
+    method: 'get',
+    impl: sendResolution(context, () => [
```
Comment (geoff-sheldon) [inline src/routes.js]: So, when this route gets accessed, does it call `sendResolution` with context and this function, and then call that with `req, res, next` via the library?

**Review by geoff-sheldon — APPROVED** [review]
This mostly makes sense - I need to learn more about testing, as that `=deps` bit is a little confusing, but I know it works because I ran it locally.  Added an under-the-hood question that won't really affect the status of the review.  

I dont think we need to worry about test coverage yet, since it's pretty early days, but if you want I can take a crack at adding some more (it'd be good practice!)

**Comment by hibes** [comment]
> This mostly makes sense - I need to learn more about testing, as that `=deps` bit is a little confusing, but I know it works because I ran it locally. Added an under-the-hood question that won't really affect the status of the review.

Yeah... I need to put some documentation up in `@0time/test-deps` to help illustrate what's going on there. I just wrapped a bunch of test-related gibberish into a import and force it into each test context using https://github.com/0time/npcrm-server/blob/master/test/add-deps-global.js.
 
> I dont think we need to worry about test coverage yet, since it's pretty early days, but if you want I can take a crack at adding some more (it'd be good practice!)

Yeah, I've been dabbling with test coverage, too, but you're certainly welcome to contribute however you want!

File: src/routes.js (hunk 2 above)
Comment (hibes) [reply src/routes.js]: Yes, it configures sendResolution with the app context and that function which returns constant example customer data, which can easily be swapped out for any function which returns data. The configured function it returns is a normal express route handler using args (req, res, next).

File: src/lib/send-resolution.js (hunk 1 above)
Comment (hibes) [reply src/lib/send-resolution.js]: Yes! Hmm, context is unused here, I should fix that.

Maybe fnMayPromise should have a signature like:
`fnMayPromise({context, req, res})` and that function can decide which of those args it wants to use.

**Pull request merged**